### PR TITLE
fix(cli/server): serve command expected positional args

### PR DESCRIPTION
**Pull request by ralgozino**
Expect exactly one positional argument for the serve command with the path to the config file.

**What type of PR is this?**

bug

**Which issue does this PR fix**:

There's no reported issue AFAIK.

**What does this PR do / Why do we need it**:

This PR adds a check to the `serve` command to expect exactly one positional argument with the config file path. Improving the user experience.

**If an issue # is not available please add repro steps and logs showing the issue**:

Running `zot serve` (without passing a config file path) results in an index out-of-range error:

```
$ ./bin/zot-darwin-arm64 serve
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
zotregistry.dev/zot/pkg/cli/server.NewServerRootCmd.newServeCmd.func2(0x14002026800?, {0x10aede5e0, 0x0, 0x105f211c0?})
	zotregistry.dev/zot/pkg/cli/server/root.go:65 +0x1b0
github.com/spf13/cobra.(*Command).execute(0x14000354608, {0x10aede5e0, 0x0, 0x0})
	github.com/spf13/cobra@v1.8.0/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x14000354308)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(0x10aab7538?)
	github.com/spf13/cobra@v1.8.0/command.go:1039 +0x1c
main.main()
	zotregistry.dev/zot/cmd/zot/main.go:10 +0x20
```

As a new user, I found this error to be misleading. I thought I had some error on the json file instead (I assumed that the `config.json` file would be used by default).

**Testing done on this change**:

With the change included in this PR applied, this is the output you get instead:

```
$ ./bin/zot-darwin-arm64-minimal serve
Error: accepts 1 arg(s), received 0
Usage:
  zot serve <config> [flags]

Aliases:
  serve, serve

Flags:
  -h, --help   help for serve
```

Passing 1 argument works as expected:

```
$ ./bin/zot-darwin-arm64-minimal serve non-existent.json
{"level":"error","error":"open non-existent.json: no such file or directory","path":"non-existent.json","time":"2024-04-10T10:42:52+02:00","message":"failed to read configuration"}
Error: open non-existent.json: no such file or directory
Usage:
  zot serve <config> [flags]

Aliases:
  serve, serve

Flags:
  -h, --help   help for serve
```

**Automation added to e2e**:

I've added a unit test that checks for an error when not passing the expected positional argument.

Note: I tried running `make` locally, but it is failing on my machine even when running it on the `main` branch. I don't expect this change to impact other parts of the code.

**Will this break upgrades or downgrades?**

No

**Does this PR introduce any user-facing change?**:

```release-note
Running `zot serve` without the path to a config file won't result in an index out-of-range error anymore, it will show the help message of the command instead instructing to add the missing argument.
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — DISMISSED** [review]

File: pkg/cli/server/root_test.go [hunk 1]
```diff
@@ -48,6 +48,12 @@ func TestServe(t *testing.T) {
 	})
 
 	Convey("Test serve config", t, func(c C) {
```
Comment (rchincha) [inline pkg/cli/server/root_test.go:50]: Also add a test case for more than one arg

File: pkg/cli/server/root_test.go [hunk 2]
```diff
@@ -1256,11 +1262,11 @@ storage:
 		err = tmpCredsFile.Close()
 		So(err, ShouldBeNil)
 
-		content = []byte(fmt.Sprintf(`{ "distSpecVersion": "1.1.0-dev", 
-			"storage": { "rootDirectory": "/tmp/zot" }, "http": { "address": "127.0.0.1", "port": "8080", 
-			"auth": { "ldap": { "credentialsFile": "%v", "address": "ldap.example.org", "port": 389, 
-			"startTLS": false, "baseDN": "ou=Users,dc=example,dc=org", 
-			"userAttribute": "uid", "userGroupAttribute": "memberOf", "skipVerify": true, "subtreeSearch": true }, 
+		content = []byte(fmt.Sprintf(`{ "distSpecVersion": "1.1.0-dev",
```
Comment (rchincha) [inline pkg/cli/server/root_test.go:1265]: Are these whitespace cleanups? Would move this to a separate commit

**Review by rchincha — COMMENTED** [review]
@ralgozino pls also GPG sign your commits.
https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by ralgozino** [comment]
> @ralgozino pls also GPG sign your commits.
> https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

Interesting, I just realized that I have it setup only for my work email address, I will fix it.

Is it enough to GPG sign or should I also sign off the commit? (I see the check for sign off in Ci is failing).

I guess I will have to force push to overwrite the commit

**Comment by rchincha** [comment]
> > @ralgozino pls also GPG sign your commits.
> > https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits
> 
> Interesting, I just realized that I have it setup only for my work email address, I will fix it.
> 
> Is it enough to GPG sign or should I also sign off the commit? (I see the check for sign off in Ci is failing).
> 
> I guess I will have to force push to overwrite the commit

You would have to sign and sign-off also.

File: pkg/cli/server/root_test.go (hunk 2 above)
Comment (ralgozino) [reply pkg/cli/server/root_test.go]: I've removed these changes

**Comment by ralgozino** [comment]
I should have addressed all your suggestions @rchincha, let me know if something else needs to be done.

Thank you for the review.

File: pkg/cli/server/root_test.go (hunk 1 above)
Comment (ralgozino) [reply pkg/cli/server/root_test.go:50]: Added the test case for 2 args

**Review by rchincha — APPROVED** [review]
lgtm

**Comment by rchincha** [comment]
@ralgozino thanks for your PR.
